File: qqbot-module-operations/src/main/java/com/mieai/qqbot/admin/events/PluginDeliveryController.java
package com.mieai.qqbot.admin.events;

import java.util.Objects;
import org.springframework.http.CacheControl;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping({"/api/events/plugin-deliveries", "/api/plugin-deliveries"})
public class PluginDeliveryController {
    private final PluginDeliveryAdministrationService service;

    public PluginDeliveryController(PluginDeliveryAdministrationService service) {
        this.service = Objects.requireNonNull(service, "service must not be null");
    }

    @GetMapping
    public ResponseEntity<PluginDeliveryPageResponse> list(
            @RequestParam(defaultValue = "50") String limit,
            @RequestParam(required = false) String cursor,
            @RequestParam(required = false) String query,
            @RequestParam(required = false) String bindingId,
            @RequestParam(required = false) String status) {
        return noStore(service.list(limit, cursor, query, bindingId, status, false));
    }

    @GetMapping("/stats")
    public ResponseEntity<PluginDeliveryQueueStatsResponse> stats() {
        return noStore(service.statistics());
    }

    @GetMapping("/{id}")
    public ResponseEntity<PluginDeliveryDetailResponse> get(@PathVariable String id) {
        return noStore(service.get(id, false));
    }

    static <T> ResponseEntity<T> noStore(T body) {
        return ResponseEntity.ok().cacheControl(CacheControl.noStore()).body(body);
    }
}

File: qqbot-module-operations/src/main/kotlin/com/mieai/qqbot/admin/events/PluginDeliveryController.kt
package com.mieai.qqbot.admin.events

import org.springframework.http.CacheControl
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/events/plugin-deliveries", "/api/plugin-deliveries")
class PluginDeliveryController(
    private val service: PluginDeliveryAdministrationService,
) {
    @GetMapping
    fun list(
        @RequestParam(defaultValue = "50") limit: String,
        @RequestParam(required = false) cursor: String?,
        @RequestParam(required = false) query: String?,
        @RequestParam(required = false) bindingId: String?,
        @RequestParam(required = false) status: String?,
    ): ResponseEntity<PluginDeliveryPageResponse> =
        noStore(service.list(limit, cursor, query, bindingId, status, false))

    @GetMapping("/stats")
    fun stats(): ResponseEntity<PluginDeliveryQueueStatsResponse> = noStore(service.statistics())

    @GetMapping("/{id}")
    fun get(@PathVariable id: String): ResponseEntity<PluginDeliveryDetailResponse> =
        noStore(service.get(id, false))

    companion object {
        fun <T> noStore(body: T): ResponseEntity<T> = ResponseEntity.ok()
            .cacheControl(CacheControl.noStore())
            .body(body)
    }
}
